libsolidity/formal/SMTChecker.cpp

```diff
@@ -681,11 +746,15 @@ void SMTChecker::checkCondition(
 			expressionNames.push_back(_additionalValueName);
 		}
 		for (auto const& var: m_variables)
-			if (knownVariable(*var.first))
-			{
-				expressionsToEvaluate.emplace_back(currentValue(*var.first));
-				expressionNames.push_back(var.first->name());
-			}
+		{
+			expressionsToEvaluate.emplace_back(currentValue(*var.first));
+			expressionNames.push_back(var.first->name());
+		}
+		for (auto const& var: m_specialVariables)
+		{
+			expressionsToEvaluate.emplace_back(var.second->currentValue());
```

Comment:
We should somehow reduce the size of the counterexamples.

Reply:
We currently ask for a value for every state variable, local variable, and special variable/function used. I think reducing it would require some non trivial analysis.